README.md

```diff
@@ -105,6 +106,46 @@ This command runs all available unit tests.
 
 ## Experimental features
 
+### `pdk console` command
+The pdk console command executes a session of the puppet debugger when inside a module and allows for exploration of puppet code.  See the official [puppet debugger site](https://www.puppet-debugger.com) for more info and the official docs [site here.](https://docs.puppet-debugger.com)
+
+To use, execute `pdk console` from inside your module directory.  You can also supply the `--puppet-version` or `--pe-version` or `--puppet-dev` to swap out the puppet version when using the console.
+
+Example (from within a module):
+
+* `pdk console --puppet-version=5`
+* `pdk console --pe-version=2018.1`
+
+The `pdk console` command will also pass through any puppet debugger arguments you wish to use.
+
+Example:
+
+* `pdk console  --no-facterdb`
+* `pdk console --play https://gist.github.com/logicminds/4f6bcfd723c92aad1f01f6a800319fa4`
+* `pdk console -e "md5('sdfasdfasdf')" --run-once --quiet`
+
+Use `pdk console -h` for a further explanation of pass through arguments.
+
+If you receive the following error you do not have the puppet-debugger gem installed.
+
+```
+pdk console -h
+Error: Unknown Puppet subcommand 'debugger'
+See 'puppet help' for help on available puppet subcommands
```

Comment:
`See 'puppet help' for help on available puppet subcommands` seems the wrong command here.  Should be `pdk help`?

Reply:
Because option parsing is disabled there is not a good way to intercept the help command as parsing is either on or off.  As a result the help or -h is sent to the puppet debugger face.  However, because the debugger is not installed on this system you get that response.  This error should be caught.  Assume you have the latest code?

Reply:
I did force push almost 20 times so you might be running older code.

```
pdk help console
NAME
    console - (Experimental) Start a session of the puppet debugger console.

USAGE
    pdk console [console_options]

DESCRIPTION
    The pdk console runs a interactive session of the puppet debugger tool to
    test out snippets of code, run language evaluations, datatype prototyping
    and much more.  A virtual playground for your puppet code! For usage
    details see the puppet debugger docs at https://docs.puppet-debugger.com.

OPTIONS
       --pe-version=<value>          Puppet Enterprise version to run tests
                                     or validations against.
       --puppet-dev                  When specified, PDK will validate or
                                     test against the current Puppet source
                                     from github.com. To use this option, you
                                     must have network access to
                                     https://github.com.
       --puppet-version=<value>      Puppet version to run tests or
                                     validations against.

OPTIONS FOR PDK
    -d --debug                       Enable debug output.
    -f --format=<value>              Specify desired output format. Valid
                                     formats are 'junit', 'text'. You may
                                     also specify a file to which the
                                     formatted output is sent, for example:
                                     '--format=junit:report.xml'. This option
                                     may be specified multiple times if each
                                     option specifies a distinct target file.
    -h --help                        Show help for this command.
       --version                     Show version of pdk.
 ~/nwops/gofish   master ●  pdk console -h
pdk (FATAL): Your Gemfile is missing the puppet-debugger gem.  You can add the missing gem
by updating your /Users/cosman/nwops/gofish/.sync.yml file with the following
and running pdk update.

Gemfile:
  required:
    ":development":
      - gem: puppet-debugger
        version: "~> 0.14"
```